src/plugins/intel_cpu/src/nodes/matmul.cpp

```diff
@@ -490,28 +509,10 @@ void MatMul::prepareParams() {
     if (selected_pd == nullptr)
         IE_THROW()  << errorPrefix << " did not set preferable primitive descriptor";
 
-    DnnlMemoryDescPtr src0TransposedDesc;
-    DnnlMemoryDescPtr src1TransposedDesc;
-
-    AttrPtr attr;
-
     if (isDynamicNode()) {
-        attr = initPrimitiveAttr(dstMemPtr->getStaticDims());
-
-        const auto& src0Desc = src0MemPtr->getDesc();
-        const auto& src1Desc = src1MemPtr->getDesc();
-
-        auto src0Shape = src0Desc.getShape();
-        auto src0Strides = getStridesAndModifyShape(src0Shape, transposeIn[0]);
-        src0TransposedDesc = std::make_shared<DnnlBlockedMemoryDesc>(src0Desc.getPrecision(), src0Shape, src0Strides);
-
-        auto src1Shape = src1Desc.getShape();
-        auto src1Strides = getStridesAndModifyShape(src1Shape, transposeIn[1]);
-        src1TransposedDesc = std::make_shared<DnnlBlockedMemoryDesc>(src1Desc.getPrecision(), src1Shape, src1Strides);
+        setPostOps(*attr, dstMemPtr->getStaticDims(), true);
```

Comment:
Your idea is great, but there remains another problem, the MatmulKey is changed not to use the src0TransposedDesc & src1TransposedDesc. In static case, I cannot re-use the src0TransposedDesc & src0TransposedDesc since it's not part of the key.